### PR TITLE
[test] Modernise magma tests

### PR DESCRIPTION
**Pull request by victorusu**

### CODE REVIEW

File: cscs-checks/libraries/magma/magma_checks.py [hunk 1]
```diff
@@ -7,29 +7,36 @@
...
-        self.prebuild_cmds = ['patch < patch.txt']
+    @run_after('init')
+    def set_build_system_opts(self):
         self.build_system = 'Make'
-        self.valid_prog_environs = ['builtin']
-        self.build_system.makefile = 'Makefile_%s' % subtest
+        self.build_system.makefile = 'Makefile_%s' % self.subtest
         # Compile with -O0 since with a higher level the compiler seems to
-        # optimise something away
+        # optimise stuff away
```
Comment (victorusu) [inline cscs-checks/libraries/magma/magma_checks.py:26]: @ajocksch, is it bad that the compiler is optimizing stuff away? Does it break the build?

**Comment by victorusu** [comment]
@jenkins-cscs retry all

File: cscs-checks/libraries/magma/magma_checks.py [hunk 2]
```diff
@@ -7,29 +7,36 @@
...
+    maintainers = ['AJ', 'SK']
+    tags = {'scs', 'production', 'maintenance'}
 
-        self.prebuild_cmds = ['patch < patch.txt']
+    @run_after('init')
+    def set_build_system_opts(self):
         self.build_system = 'Make'
-        self.valid_prog_environs = ['builtin']
-        self.build_system.makefile = 'Makefile_%s' % subtest
+        self.build_system.makefile = 'Makefile_%s' % self.subtest
```
Comment (teojgo) [inline cscs-checks/libraries/magma/magma_checks.py:24]:
```suggestion
        self.build_system.makefile =f 'Makefile_{self.subtest}'
```

File: cscs-checks/libraries/magma/magma_checks.py [hunk 3]
```diff
@@ -7,29 +7,36 @@
...
         self.build_system.ldflags = ['-lcusparse', '-lcublas', '-lmagma',
                                      '-lmagma_sparse']
-        self.executable = './testing_' + subtest
-        self.modules = ['magma']
-        self.maintainers = ['AJ', 'SK']
-        self.tags = {'scs', 'production', 'maintenance'}
-        if subtest == 'cblas_z':
+        self.executable = './testing_' + self.subtest
+
+    @run_after('init')
```
Comment (teojgo) [inline cscs-checks/libraries/magma/magma_checks.py:33]:
```suggestion
    @run_before('sanity')
```

**Review by teojgo — CHANGES_REQUESTED** [review]

File: cscs-checks/libraries/magma/magma_checks.py [hunk 4]
```diff
@@ -7,29 +7,36 @@
...
         self.build_system.cflags = ['-O0']
         self.build_system.cxxflags = ['-O0', '-std=c++11']
         self.build_system.ldflags = ['-lcusparse', '-lcublas', '-lmagma',
                                      '-lmagma_sparse']
-        self.executable = './testing_' + subtest
-        self.modules = ['magma']
-        self.maintainers = ['AJ', 'SK']
-        self.tags = {'scs', 'production', 'maintenance'}
-        if subtest == 'cblas_z':
+        self.executable = './testing_' + self.subtest
```
Comment (vkarak) [inline cscs-checks/libraries/magma/magma_checks.py:31]:
```suggestion
        self.executable = f'./testing_{self.subtest}'
```

File: cscs-checks/libraries/magma/magma_checks.py [hunk 5]
```diff
@@ -7,29 +7,36 @@
...
+
+    @run_after('init')
+    def set_sanity_patterns(self):
+        self.sanity_patterns = sn.assert_found(r'Result = PASS', self.stdout)
```
Comment (vkarak) [inline cscs-checks/libraries/magma/magma_checks.py:35]:
```suggestion
    @sanity_function
    def assert_result(self):
        return sn.assert_found(r'Result = PASS', self.stdout)
```

File: cscs-checks/libraries/magma/magma_checks.py (hunk 3 above)
Comment (vkarak) [reply cscs-checks/libraries/magma/magma_checks.py:33]: See my suggestion  below. We should use the `@sanity_function` decorator.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Comment by teojgo** [comment]
I will fix the PR comments on this.

File: cscs-checks/libraries/magma/src/Makefile_cblas_z [hunk 6]
```diff
@@ -1,7 +1,6 @@
 all: testing_cblas_z
 
 testing_cblas_z: testing_cblas_z.o magma_util.o
-#	CC testing_cblas_z.o magma_util.o -lcusparse -lcublas /apps/daint/UES/jenkins/5.2.UP04/easybuild/software/magma/2.0.0-CrayGNU-2016.03/lib/libmagma.a /apps/daint/UES/jenkins/5.2.UP04/easybuild/software/magma/2.0.0-CrayGNU-2016.03/lib/libmagma_sparse.a -o testing_cblas_z
```
Comment (vkarak) [inline cscs-checks/libraries/magma/src/Makefile_cblas_z:4]: Perhaps remove this line?

**Review by vkarak — APPROVED** [review]
Lgtm, I have just a very  minor comment.

File: cscs-checks/libraries/magma/src/Makefile_cblas_z (hunk 6 above)
Comment (teojgo) [reply cscs-checks/libraries/magma/src/Makefile_cblas_z:4]: This is already removed

**Review by teojgo — APPROVED** [review]

**Pull request merged**